libvast/src/system/partition_transformer.cpp

```diff
@@ -95,91 +99,115 @@ void partition_transformer_state::add_slice(const table_slice& slice) {
 
 void partition_transformer_state::finalize_data() {
   // Serialize the finished indexers.
-  for (auto& [qf, idx] : indexers) {
-    data.indexer_chunks.emplace_back(qf.name(), chunkify(idx));
+  for (auto& [layout, typed_indexers] : indexers)
+    for (auto& [qf, idx] : typed_indexers)
+      data[layout].indexer_chunks.emplace_back(qf.name(), chunkify(idx));
+  for (auto& [layout, partition_data] : data) {
+    auto& mutable_synopsis = partition_data.synopsis.unshared();
+    mutable_synopsis.shrink();
+    // TODO: It would probably make more sense if the partition
+    // synopsis keeps track of offset/events internally.
+    mutable_synopsis.offset = partition_data.offset;
+    mutable_synopsis.events = partition_data.events;
   }
-  auto& mutable_synopsis = data.synopsis.unshared();
-  mutable_synopsis.shrink();
-  // TODO: It would probably make more sense if the partition
-  // synopsis keeps track of offset/events internally.
-  mutable_synopsis.offset = data.offset;
-  mutable_synopsis.events = data.events;
 }
 
 void partition_transformer_state::fulfill(
   partition_transformer_actor::stateful_pointer<partition_transformer_state>
     self,
   stream_data&& stream_data, path_data&& path_data) const {
+  auto promise = path_data.promise;
   if (self->state.stream_error) {
-    path_data.promise.deliver(self->state.stream_error);
+    promise.deliver(self->state.stream_error);
     self->quit();
     return;
   }
-  auto promise = path_data.promise;
   if (self->state.events == 0) {
-    promise.deliver(augmented_partition_synopsis{
-      .uuid = vast::uuid::nil(),
-      .stats = std::move(self->state.stats),
-      .synopsis = nullptr,
-    });
+    promise.deliver(std::vector<augmented_partition_synopsis>{});
```

Comment:
We should probably store a transformation error in the state in case that went wrong.